postprocessor/src/main/java/com/azure/autorest/postprocessor/util/PartialUpdateHandler.java

```diff
@@ -160,11 +162,109 @@ private static String handlePartialUpdateForClassOrInterfaceFile(String generate
+        NodeList<ModuleDirective> providesDirectivesForExistingFile = new NodeList<>();
+        for (Iterator<ModuleDirective> it = directivesForExistingFile.stream().iterator(); it.hasNext(); ) {
+            ModuleDirective directive = it.next();
+            if (directive.isModuleRequiresDirective()) {
+                requiresDirectivesForExistingFile.add(directive);
+            }
+            if (directive.isModuleExportsDirective()) {
+                exportsDirectivesForExistingFile.add(directive);
+            }
+            if (directive.isModuleOpensDirective()) {
+                opensDirectivesForExistingFile.add(directive);
+            }
+            if (directive.isModuleUsesDirective()) {
+                usesDirectivesForExistingFile.add(directive);
+            }
+            if (directive.isModuleProvidesDirective()) {
+                providesDirectivesForExistingFile.add(directive);
+            }
+        }
+
+        NodeList<ModuleDirective> requiresDirectiveNodeList = mergeDirectiveNodeList(requiresDirectivesForGeneratedFile, requiresDirectivesForExistingFile);
+        NodeList<ModuleDirective> exportsDirectiveNodeList = mergeDirectiveNodeList(exportsDirectivesForGeneratedFile, exportsDirectivesForExistingFile);
+        NodeList<ModuleDirective> opensDirectiveNodeList = mergeDirectiveNodeList(opensDirectivesForGeneratedFile, opensDirectivesForExistingFile);
+        NodeList<ModuleDirective> usesDirectiveNodeList = mergeDirectiveNodeList(usesDirectivesForGeneratedFile, usesDirectivesForExistingFile);
+        NodeList<ModuleDirective> providesDirectiveNodeList = mergeDirectiveNodeList(providesDirectivesForGeneratedFile, providesDirectivesForExistingFile);
+
+        NodeList<ModuleDirective> moduleDirectives = new NodeList<>();
+        moduleDirectives.addAll(requiresDirectiveNodeList);
+        moduleDirectives.addAll(exportsDirectiveNodeList);
+        moduleDirectives.addAll(opensDirectiveNodeList);
+        moduleDirectives.addAll(usesDirectiveNodeList);
+        moduleDirectives.addAll(providesDirectiveNodeList);
+
+        ModuleDeclaration moduleDeclaration = compilationUnitForGeneratedFile.getModule().get();
+        moduleDeclaration.setDirectives(moduleDirectives);
+
+        compilationUnitForGeneratedFile.setModule(moduleDeclaration);
+
+        return compilationUnitForGeneratedFile.toString();
+    }
+
+    private static NodeList<ModuleDirective> mergeDirectiveNodeList(NodeList<ModuleDirective> list1, NodeList<ModuleDirective> list2) {
+        NodeList<ModuleDirective> res = new NodeList<>();
+        res.addAll(list1);
+        for (ModuleDirective directive2 : list2) {
+            boolean isInList1 = false;
+            for (ModuleDirective directive1 : list1) {
+                if(directive1.getTokenRange().isPresent() && directive2.getTokenRange().isPresent() &&
+                        directive1.getTokenRange().get().toString().equals(directive2.getTokenRange().get().toString())) {
```

Comment:
So for now, it does not support merge of "open to A,B" and "open to B,A".
That might be fine.

Does it handle a slight formatting difference e.g. extra space or extra new-line in existing file? E.g. "export  A" and "export A"

Reply:
> So for now, it does not support merge of "open to A,B" and "open to B,A". That might be fine.
> 
> Does it handle a slight formatting difference e.g. extra space or extra new-line in existing file? E.g. "export A" and "export A"

1. It does not support merge of "open to A,B" and "open to B,A"
2. Yes, it will ignore empty line or space, I added a test to cover this. https://github.com/Azure/autorest.java/pull/1451/commits/c49b743aa95863415816833f76d0448679a4c9cd

Reply:
Could you update your test to space or new-line in the middle?
E.g.
```
exports 
    com.azure.communication.phonenumbersdemo;
```